dpe/src/commands/derive_context.rs

```diff
@@ -202,6 +211,18 @@ impl CommandExecution for DeriveChildCmd {
             dpe.generate_new_handle(env)?
         };
 
+        let mut allow_ca = self.allows_ca();
+        let mut allow_x509 = self.allows_x509();
+        let mut uses_internal_input_info = self.uses_internal_info_input();
+        let mut uses_internal_input_dice = self.uses_internal_dice_input();
+        // if recursive flag is set, ignore other flags
+        if self.is_recursive() {
+            allow_ca = dpe.contexts[parent_idx].allow_ca();
+            allow_x509 = dpe.contexts[parent_idx].allow_x509();
+            uses_internal_input_info = dpe.contexts[parent_idx].uses_internal_input_info();
+            uses_internal_input_dice = dpe.contexts[parent_idx].uses_internal_input_dice();
+        }
+
         // Create a temporary context to mutate so that we avoid mutating internal state upon an error.
         let mut tmp_child_context = Context::new();
         tmp_child_context.activate(&ActiveContextArgs {
```

Comment:
I think if `is_recursive`, we should be basically doing the `extend_tci` behavior (update parent rather than making a new child context)?

Reply:
Done.